crates/relayer-types/src/core/ics24_host/identifier.rs

```diff
@@ -68,8 +68,20 @@ impl ChainId {
 
     /// Extract the chain name from this chain identifier. The chain name
     /// consists of the first part of the identifier, before the dash.
+    /// If the value following the dash is not an integer, or if there are no
+    /// dashes, the entire chain ID will be returned.
     pub fn name(&self) -> String {
```

Comment:
I think we can change this to return a string slice to avoid allocations, but it's not strictly necessary.

```suggestion
    pub fn name(&self) -> &str {
```